src/clients/java/src/client.zig

```diff
@@ -110,19 +115,58 @@ const ReflectionHelper = struct {
         assert(env.hasPendingException());
     }
 
-    pub fn buffer(env: *jui.JNIEnv, this_obj: jui.jobject) ?[]u8 {
+    pub fn get_send_buffer(env: *jui.JNIEnv, this_obj: jui.jobject) ?[]u8 {
         assert(this_obj != null);
-        assert(request_buffer_field_id != null);
-        assert(request_buffer_len_field_id != null);
+        assert(request_send_buffer_field_id != null);
+        assert(request_send_buffer_len_field_id != null);
 
-        var buffer_obj = env.getField(.object, this_obj, request_buffer_field_id) orelse return null;
+        var buffer_obj = env.getField(.object, this_obj, request_send_buffer_field_id) orelse
+            return null;
         defer env.deleteReference(.local, buffer_obj);
 
-        var buffer_len = env.getField(.long, this_obj, request_buffer_len_field_id);
-        var address = env.getDirectBufferAddress(buffer_obj);
+        var buffer_len = env.getField(.long, this_obj, request_send_buffer_len_field_id);
+        var buffer: []u8 = env.getDirectBufferAddress(buffer_obj);
 
         // The buffer can be larger than the actual message content,
-        return address[0..@intCast(usize, buffer_len)];
+        return buffer[0..@intCast(usize, buffer_len)];
+    }
+
+    pub fn set_reply_buffer(env: *jui.JNIEnv, this_obj: jui.jobject, reply: []const u8) void {
+        assert(this_obj != null);
+        assert(request_reply_buffer_field_id != null);
+        assert(reply.len > 0);
+
+        var reply_buffer_obj = env.newPrimitiveArray(
+            .byte,
+            @intCast(c_int, reply.len),
+        ) catch {
+            // Cannot allocate an array, it's likely the JVM has run out of resources
+            // Printing the buffer size here just to help diagnosing how much memory was required
+            env.describeException();
+            log.err("Unexpected error allocating a new array len={}", .{reply.len});
+            @panic("JNI: Unexpected error creating a new array.");
```

Comment:
Would it be possible to rethrow the exception to the JVM here? Crashing the whole process on OOM feels suboptimal (in contrast, crashing after `setPrimitiveArrayRegion` feels fine, as an excetpion there means a programming error in our code).

Anywhay, this is unrelated to the PR in question, as we had similar behavior before.

Reply:
On the JNI side, the only place that throws exceptions is the initialization (for validating the parameters).
Panicking the entire process feels suboptimal and can be inconvenient for the application's developers.

a) Environment errors, like OOM. We could try to throw an exception, but it can fail due to OOM anyway 🤪, so we don't bother with that and panic;

b) Unexpected conditions in our code are treated with `assert`.

c) We cannot throw exceptions inside the native callback.

(...)

I'll review this strategy for minimizing the places where we panic at library code in general.